src/framework/components/element/component.js

```diff
@@ -674,27 +679,33 @@ class ElementComponent extends Component {
      * The position of the pivot of the component relative to its anchor. Each value ranges from 0
      * to 1 where [0,0] is the bottom left and [1,1] is the top right.
      *
-     * @type {Vec2}
+     * @example
+     * pc.app.root.findByName("Inventory").element.pivot = [Math.random() * 0.1, Math.random() * 0.1];
+     * @example
+     * pc.app.root.findByName("Inventory").element.pivot = new pc.Vec2(Math.random() * 0.1, Math.random() * 0.1);
+     *
+     * @type {Vec2 | number[]}
      */
     set pivot(value) {
-        const prevX = this._pivot.x;
-        const prevY = this._pivot.y;
+        const { _pivot, _margin } = this;
```

Comment:
why local variables with _ ? I'd remove it.

Reply:
Do you mean like `const { _pivot: pivot, _margin: margin } = this;` instead?

Reply:
yep, that would do.

Reply:
I just realized its also possible to just write `const { pivot, margin } = this;`, since `get pivot()` is just returning `_pivot` anyway, which would decrease the amount of code here. Is that okay with you?

Reply:
totally